aws/resource_aws_workspaces_workspace.go

```diff
@@ -0,0 +1,498 @@
+
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{
+			workspaces.WorkspaceStateUpdating,
+		},
+		Target: []string{
+			workspaces.WorkspaceStateAvailable,
+			workspaces.WorkspaceStateStopped,
+		},
+		Refresh:      workspaceRefreshStateFunc(conn, d.Id()),
+		PollInterval: 30 * time.Second,
+		Timeout:      10 * time.Minute,
+	}
+
+	_, err = stateConf.WaitForState()
+	if err != nil {
+		return fmt.Errorf("workspace %q %s property was not modified: %s", d.Id(), p, err)
+	}
+	log.Printf("[DEBUG] Workspace %q %s property is modified", d.Id(), p)
+
+	return nil
+}
+
+func workspaceRefreshStateFunc(conn *workspaces.WorkSpaces, workspaceID string) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		resp, err := conn.DescribeWorkspaces(&workspaces.DescribeWorkspacesInput{
+			WorkspaceIds: []*string{aws.String(workspaceID)},
+		})
+		if err != nil {
+			return nil, workspaces.WorkspaceStateError, err
+		}
+		if len(resp.Workspaces) == 0 {
+			return resp, workspaces.WorkspaceStateTerminated, nil
+		}
+		workspace := resp.Workspaces[0]
+		return workspace, aws.StringValue(workspace.State), nil
+	}
+}
+
+func expandWorkspaceProperties(properties []interface{}) *workspaces.WorkspaceProperties {
+	log.Printf("[DEBUG] Expand Workspace properties: %+v ", properties)
+
+	if len(properties) == 0 || properties[0] == nil {
+		return nil
+	}
+
+	p := properties[0].(map[string]interface{})
+
+	return &workspaces.WorkspaceProperties{
+		ComputeTypeName:                     aws.String(p["compute_type_name"].(string)),
+		RootVolumeSizeGib:                   aws.Int64(int64(p["root_volume_size_gib"].(int))),
+		RunningMode:                         aws.String(p["running_mode"].(string)),
+		RunningModeAutoStopTimeoutInMinutes: aws.Int64(int64(p["running_mode_auto_stop_timeout_in_minutes"].(int))),
+		UserVolumeSizeGib:                   aws.Int64(int64(p["user_volume_size_gib"].(int))),
+	}
+}
+
+func flattenWorkspaceProperties(properties *workspaces.WorkspaceProperties) []map[string]interface{} {
+	log.Printf("[DEBUG] Flatten workspace properties: %+v ", properties)
+
+	if properties == nil {
+		return []map[string]interface{}{}
+	}
+
+	return []map[string]interface{}{
+		{
+			"compute_type_name":                         aws.StringValue(properties.ComputeTypeName),
+			"root_volume_size_gib":                      int(aws.Int64Value(properties.RootVolumeSizeGib)),
+			"running_mode":                              aws.StringValue(properties.RunningMode),
+			"user_volume_size_gib":                      int(aws.Int64Value(properties.UserVolumeSizeGib)),
+			"running_mode_auto_stop_timeout_in_minutes": int(aws.Int64Value(properties.RunningModeAutoStopTimeoutInMinutes)),
+		},
+	}
+}
+
+// NOTE: We need custom update tags function, because conn.CreateTags doesn't delete old tags
+func WorkspacesUpdateTags(conn *workspaces.WorkSpaces, identifier string, oldTagsMap interface{}, newTagsMap interface{}) error {
```

Comment:
this should solve it: https://github.com/terraform-providers/terraform-provider-aws/pull/13089

Reply:
Make sense to me, thanks for standardization 🙇  I will accommodate autogen code for sure.